src/layout/GridList/index.tsx

```diff
@@ -1,42 +1,50 @@
-import React, { PureComponent } from 'react';
+import React from 'react';
 import styled from 'styled-components';
 
 import { media } from '../../core/BreakPoints';
 import { HTMLDivProps } from '../../interfaces/props';
-import { createUniqIDGenerator } from '../../utils/createUniqIDGenerator';
+import { defaultKeyExtractor, memoComponent } from '../../utils';
 
 const sizeToPercent = (column?: number) => 100 / (column || 1);
-const generateID = createUniqIDGenerator('grid-list-');
 
-type Column = 1 | 2 | 3 | 4 | 6 | 12;
-interface Props {
-  items: any[] | ReadonlyArray<any>;
-  renderItem: any;
-  smColumn: Column;
-  lgColumn?: Column;
+export type Column = 1 | 2 | 3 | 4 | 6 | 12;
+export interface GridProps<T> {
   className?: string;
   divAttributes?: HTMLDivProps;
+  items: T[] | ReadonlyArray<T>;
+  keyExtractor?: (item: T, index: number) => string | number;
+  lgColumn?: Column;
+  renderItem: any;
+  smColumn: Column;
 }
 
-export class GridList extends PureComponent<Props> {
-  public render() {
-    const { items, renderItem, smColumn, lgColumn, className, divAttributes } = this.props;
-
+export const GridList = memoComponent(
+  <T extends any>({
```

Comment:
앗 이거 내가 정확히 리뷰를 전달하지 못했다

1. items에 어떤 형태의 배열이든 들어올 수 있게
```suggestion
  <T>({
```
이렇게 해주고,

2. 만약 breaking change를 줄이고 싶다면 (T가 생기면서 기존에 사용하던 모든 GridList에 T를 설정해줘야하니)
```suggestion
  <T = any>({
```

이렇게 기본값을 지정하자는 얘기여쓰!

Reply:
근데 어쩌면 기본값을 지정하지 않아도, items를 통해서 T의 타입이 알아서 추론 될 수 있을 듯?